src/azul/plugins/metadata/hca/transform.py

```diff
@@ -757,6 +758,7 @@ def _file(self, file: api.File, related_files: Iterable[api.File] = ()) -> Mutab
                 } if isinstance(file, api.SequenceFile) else {
                 }
             ),
+            'matrix_cell_count': getattr(file, 'matrix_cell_count', None),
```

Comment:
on ln 73 under the `FileAggregator` class it appears that the `matrix_cell_count` is always expected to be present for a given entity... (accessed with `[]`).

Is something different here?

Reply:
Yes. Here in `transform.py` the method `_file` processes many types of files (eg. api.SequenceFile, api.SupplementaryFile). The field `matrix_cell_count` only exists in api.AnalysisFile so the default `None` is used with `getattr` to handle the case when a file without the `matrix_cell_count` field is processed, and to ensure all inner entities returned by `_file` will contain the `matrix_cell_count` field.

When the entity is being aggregated by the `FileAggregator` since it is guaranteed the entity will have the `matrix_cell_count` field we can access it directly with `[]` and when processed by the `SumAccumulator` any `None` values will be ignored.